recycler/src/main/java/com/elpassion/android/commons/recycler/RecyclerViewCompositeAdapter.kt

```diff
@@ -1,27 +1,25 @@
 package com.elpassion.android.commons.recycler
 
 import android.support.v7.widget.RecyclerView
-import android.view.LayoutInflater
+import android.support.v7.widget.RecyclerView.ViewHolder
 import android.view.ViewGroup
 import com.elpassion.android.commons.recycler.components.ItemsStrategy
 import com.elpassion.android.commons.recycler.components.base.ItemAdapter
+import com.elpassion.android.view.inflate
 
-class RecyclerViewCompositeAdapter<T : ItemAdapter<out RecyclerView.ViewHolder>>(
+class RecyclerViewCompositeAdapter<T : ItemAdapter>(
         private val itemsStrategy: ItemsStrategy<T>,
         private val getItemIdentifier: (position: Int) -> Long = { 0L },
-        init: (RecyclerViewCompositeAdapter<T>.() -> Unit) = {}) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+        init: (RecyclerViewCompositeAdapter<T>.() -> Unit) = {}) : RecyclerView.Adapter<ViewHolder>() {
 
     init {
         init()
     }
 
-    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
-        val itemView = LayoutInflater.from(parent.context).inflate(viewType, parent, false)
-        return allItems().first { it.viewType == viewType }.onCreateViewHolder(itemView)
-    }
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) = object : ViewHolder(parent.inflate(viewType)) {}
```

Comment:
And what is the problem? We don't need any special name for it. You were creating the same number of view holders here too, but with additional unnecessary operations (getting all adapters and iterating over it all)